planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileCopy.java

```diff
@@ -0,0 +1,236 @@
+package com.onthegomap.planetiler.archive;
+
+import com.onthegomap.planetiler.config.Arguments;
+import com.onthegomap.planetiler.config.CommonConfigs;
+import com.onthegomap.planetiler.stats.Counter;
+import com.onthegomap.planetiler.stats.ProcessInfo;
+import com.onthegomap.planetiler.stats.ProgressLoggers;
+import com.onthegomap.planetiler.stats.Stats;
+import com.onthegomap.planetiler.util.Gzip;
+import com.onthegomap.planetiler.util.Hashing;
+import com.onthegomap.planetiler.worker.WorkerPipeline;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalLong;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility to copy/convert tiles and metadata from one archive into another.
+ * <p>
+ * Example usages:
+ *
+ * <pre>
+ * --input=tiles.mbtiles --output=tiles.mbtiles
+ * --input=tiles.mbtiles --output=tiles.pmtiles --skip_empty=false
+ * --input=tiles.pmtiles --output=tiles.mbtiles
+ * --input=tiles.mbtiles --output=tiles/
+ * --input=tiles.mbtiles --output=tiles.json --out_tile_compression=gzip
+ * --input=tiles.mbtiles --output=tiles.csv --out_tile_compression=none
+ * --input=tiles.mbtiles --output=tiles.proto
+ * </pre>
+ */
+public class TileCopy {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TileCopy.class);
+
+  private final TileCopyConfig config;
+
+  private final Counter.MultiThreadCounter tilesWrittenOverall = Counter.newMultiThreadCounter();
+
+
+  TileCopy(TileCopyConfig config) {
+    this.config = config;
+  }
+
+  public void run() throws IOException {
+
+    if (!config.inArchive().exists()) {
+      throw new IllegalArgumentException("the input archive does not exist");
+    }
+
+    config.outArchive().setup(config.force(), config.append(), config.tileWriterThreads());
+
+    final var loggers = ProgressLoggers.create()
+      .addRateCounter("tiles", tilesWrittenOverall::get);
+
+    try (
+      var reader = TileArchives.newReader(config.inArchive(), config.inArguments());
+      var writer = TileArchives.newWriter(config.outArchive(), config.outArguments())
+    ) {
+
+      final TileArchiveMetadata inMetadata = getInMetadata(reader);
+      final TileArchiveMetadata outMetadata = getOutMetadata(inMetadata);
+
+      writer.initialize();
+      try (
+        var rawTiles = reader.getAllTiles();
+        var it = config.skipEmpty() ? rawTiles.filter(t -> t.bytes() != null && t.bytes().length > 0) : rawTiles
+      ) {
+
+        final var tileConverter = tileConverter(inMetadata.tileCompression(), outMetadata.tileCompression(), writer);
```

Comment:
The tile converter seems like it could get expensive if compressions are different. Given we are only using a read and write thread so far, might make sense to add a worker per core when the converter has something to do?

I could also see that converter doing the work to deserialize tiles and update tile stats eventually, or filter-out layers, apply some geometry transform, merge multiple input archives, etc. which would make sense in a worker-per-core model.